render-ws-java-client/src/main/java/org/janelia/render/client/MultiSemPreAlignClient.java

```diff
@@ -0,0 +1,164 @@
+
+	public static class Parameters extends CommandLineParameters {
+
+		@ParametersDelegate
+		public RenderWebServiceParameters renderWeb = new RenderWebServiceParameters();
+
+		@ParametersDelegate
+		public MatchCollectionParameters matches = new MatchCollectionParameters();
+
+		@Parameter(
+				names = "--stack",
+				description = "Name of source stack",
+				required = true)
+		public String stack;
+
+		@Parameter(
+				names = "--targetStack",
+				description = "Name of target stack",
+				required = true)
+		public String targetStack;
+
+		@Parameter(
+				names = "--completeTargetStack",
+				description = "Complete the target stack after pre-alignment",
+				arity = 0)
+		public boolean completeTargetStack = false;
+
+		@Parameter(
+				names = "--maxAllowedError",
+				description = "Max allowed error (default:10.0)"
+		)
+		public Double maxAllowedError = 10.0;
+
+		@Parameter(
+				names = "--maxIterations",
+				description = "Max iterations (default:1000)"
+		)
+		public Integer maxIterations = 1000;
+
+		@Parameter(
+				names = "--maxPlateauWidth",
+				description = "Max plateau width (default:250)"
+		)
+		public Integer maxPlateauWidth = 250;
+
+		@Parameter(
+				names = "--maxNumMatches",
+				description = "Max number of matches between mFOV layers (default:1000)"
+		)
+		public Integer maxNumMatches = 1000;
+
+		@Parameter(
+				names = "--numThreads",
+				description = "Number of threads (default:8)"
+		)
+		public Integer numThreads = 8;
+	}
+
+	public static void main(final String[] args) {
+		final ClientRunner clientRunner = new ClientRunner(args) {
+			@Override
+			public void runClient(final String[] args) throws Exception {
+
+				final Parameters parameters = new Parameters();
+				parameters.parse(args);
+
+				LOG.info("runClient: entry, parameters={}", parameters);
+
+				final MultiSemPreAlignClient client = new MultiSemPreAlignClient(parameters);
+
+				client.process();
+			}
+		};
+		clientRunner.run();
+	}
+
+
+	private final Parameters parameters;
+	private final RenderDataClient dataClient;
+
+	public MultiSemPreAlignClient(final Parameters parameters) {
+		this.parameters = parameters;
+		this.dataClient = parameters.renderWeb.getDataClient();
+	}
+
+	private void setUpTargetStack() throws IOException {
+		final StackMetaData stackMetaData = dataClient.getStackMetaData(parameters.stack);
+		dataClient.setupDerivedStack(stackMetaData, parameters.targetStack);
+	}
+
+	private void completeTargetStack() throws IOException {
+		dataClient.setStackState(parameters.targetStack, StackMetaData.StackState.COMPLETE);
+	}
+
+	private void process() throws IOException, ExecutionException, InterruptedException {
+		setUpTargetStack();
+
+		final ResolvedTileSpecsWithMatchPairs tileSpecsWithMatchPairs =
+				dataClient.getResolvedTilesWithMatchPairs(parameters.stack,
+														  null,
+														  parameters.matches.matchCollection,
+														  null,
+														  null,
+														  true);
```

Comment:
After seeing that the MultiSemPreAligner is building MFOV tiles and is only using cross matches, we should definitely break this up into separate tile spec and match data requests.  For the match match data, only request cross layer matches and submit one request per layer.  This will chunk things up into smaller individual requests and only return the match data you need.  For pre-align, maybe it is also sufficient to just request cross matches for the next layer (instead of for the next 3 to 6 layers)?

Reply:
As we discussed yesterday, the tile specs amount and the matches amount to ~60Mb and ~2Gb per stack, respectively. This can definitely be handled by one machine (even with thicker slabs and more mFOVs for future datasets). Thus, I wouldn't bother with chunking the requests, right now.
Also, this client does use same-layer matches (at least some of them) to connect multiple mFOVs.

Reply:
As discussed offline, I'll add a parameter that gives control over how many adjacent z-layers one wants to consider.